src/main/haskell/language-kore/app/parser/Main.hs

```diff
@@ -2,102 +2,135 @@ module Main where
 
 import           Data.Kore.ASTVerifier.DefinitionVerifier
 import           Data.Kore.Error
-import           Data.Kore.Parser.Parser
+import           Data.Kore.Parser.Parser                  (fromKore)
+import           Data.Kore.AST.Sentence                   (KoreDefinition)
 
 import           Control.Exception                        (evaluate)
 import           Control.Monad                            (when)
 import           System.Clock                             (Clock (Monotonic),
                                                            diffTimeSpec,
                                                            getTime)
-import           System.Environment                       (getArgs)
 
-data CommandLineFlags = CommandLineFlags
-    { commandLineFlagsFileName :: !(Maybe String)
-    , commandLineFlagsVerify   :: !Bool
-    , commandLineFlagsPrint    :: !Bool
+import           Options.Applicative
+import           Data.Semigroup                           ((<>))
+
+{- 
+Main module to run kore-parser
+TODO: add command line argument tab-completion
+-}
+
+-- | Main options record    
+data KoreParserOptions = KoreParserOptions
+    { fileNameArgument  :: String -- ^ Filename to parse and verify 
+    , willPrintOption   :: Bool   -- ^ Option to print definition
+    , willVerifyOption  :: Bool   -- ^ Option to verify definition
+    , willChkAttrOption :: Bool   -- ^ Option to check attributes during verification
     }
 
-usage :: String
-usage =
-    "Usage: kore-parser [--[no]verify] [--[no]print] fileName"
-
--- TODO(virgil): Use a generic command line parsing library instead.
-parseCommandLineFlags :: [String] -> CommandLineFlags
-parseCommandLineFlags [] =
-    CommandLineFlags
-        { commandLineFlagsFileName = Nothing
-        , commandLineFlagsVerify = True
-        , commandLineFlagsPrint = True
-        }
-parseCommandLineFlags (('-' : '-' : firstFlag) : commandLineReminder) =
-    addFlagAndContinue firstFlag commandLineReminder
+                    
+{-| Command Line Argument Parser -}
+commandLineParser :: Parser KoreParserOptions
+commandLineParser = KoreParserOptions
+      <$> argument str
+              ( metavar "FILE"
```

Comment:
Prefer indenting with multiples of 4 spaces. Also, maybe one more space before 'metavar'.